manager/orchestrator/taskreaper/task_reaper.go

```diff
@@ -69,14 +74,18 @@ func (tr *TaskReaper) Run(ctx context.Context) {
 		}
 
 		// On startup, scan the entire store and inspect orphaned tasks from previous life.
-		tasks, err = store.FindTasks(readTx, store.ByTaskState(api.TaskStateOrphaned))
+		orphanedTasks, err = store.FindTasks(readTx, store.ByTaskState(api.TaskStateOrphaned))
 		if err != nil {
 			log.G(ctx).WithError(err).Error("failed to find Orphaned tasks in task reaper init")
 		}
+		removeTasks, err = store.FindTasks(readTx, store.ByDesiredState(api.TaskStateRemove))
+		if err != nil {
+			log.G(ctx).WithError(err).Error("failed to find tasks with desired state REMOVE in task reaper init")
+		}
 	})
 
-	if len(tasks) > 0 {
-		for _, t := range tasks {
+	if len(orphanedTasks)+len(removeTasks) > 0 {
```

Comment:
I think this `if` statement is superfluous.

Reply:
Yeah but if I leave it in, when I come back to this code in several months and I'm like, "what doofus made this junk, who's responsible for this?" it will be a very humbling experience.